frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/FastForwardMerge.java

```diff
@@ -0,0 +1,57 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import static com.virtuslab.gitmachete.frontend.actions.backgroundables.FetchBackgroundable.LOCAL_REPOSITORY_NAME;
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.createRefspec;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.format;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import lombok.val;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.backgroundables.FetchBackgroundable;
+import com.virtuslab.gitmachete.frontend.actions.backgroundables.MergeCurrentBranchFastForwardOnlyBackgroundable;
+
+public final class FastForwardMerge {
+
+  private FastForwardMerge() {}
+
+  private static void mergeCurrentBranch(Project project,
+      GitRepository gitRepository,
+      MergeProps mergeProps) {
+    new MergeCurrentBranchFastForwardOnlyBackgroundable(project, gitRepository, mergeProps.getStayingBranch()).queue();
+  }
+
+  private static void mergeNonCurrentBranch(Project project,
+      GitRepository gitRepository,
+      MergeProps mergeProps) {
+    val stayingFullName = mergeProps.getStayingBranch().getFullName();
+    val movingFullName = mergeProps.getMovingBranch().getFullName();
+    val refspecFromChildToParent = createRefspec(stayingFullName, movingFullName, /* allowNonFastForward */ false);
+    val stayingName = mergeProps.getStayingBranch().getName();
+    val movingName = mergeProps.getMovingBranch().getName();
+    new FetchBackgroundable(
+        project,
+        gitRepository,
+        LOCAL_REPOSITORY_NAME,
+        refspecFromChildToParent,
+        getString("action.GitMachete.BaseFastForwardMergeBranchToParentAction.task-title"),
+        getString("action.GitMachete.BaseFastForwardMergeBranchToParentAction.task-subtitle"),
+        format(getString("action.GitMachete.BaseFastForwardMergeBranchToParentAction.notification.title.ff-fail"),
+            stayingName, movingName),
+        format(getString("action.GitMachete.BaseFastForwardMergeBranchToParentAction.notification.title.ff-success"),
+            stayingName, movingName))
+                .queue();
+  }
+
+  public static void perform(@Nullable String currentBranchName, String movingBranchName, Project project,
```

Comment:
Also, you don't need to pass `movingBranchName` as a separate param... it's one of the fields of `mergeProps`!

I'd even say you **shouldn't** pass such a redundant param, to avoid the bugs of passing two different values in `movingBranchName` and `mergeProps.getMovingBranch().getName()`

Reply:
and I also do not need to pass currentBranchName because I can get it from gitRepository, yes ?

Reply:
Oh! indeed